### DISCUSSION TITLE
[Other] - (Need help of jpeg color)

### DISCUSSION THREAD

**Discussion by DRCRecoveryData** [tannerhelland/PhotoDemon#370] [Q&A] [answered]
Hi,

I'm repairing some jpeg encrypted by ransomware, the color get wrong after moving the good header to corrupt jpeg, i want to know how "auto correct" color option from PhotoDemon working method from code
[Pictures.zip](https://github.com/tannerhelland/PhotoDemon/files/7189664/Pictures.zip)

**Comment by tannerhelland** [comment, marked as answer]
Hi @cuuhodrc .  Thanks for your question.

PhotoDemon's auto-correct algorithm starts with a simple white-balance adjustment.  You can see the full source code of this function here:

https://github.com/tannerhelland/PhotoDemon/blob/4e015a6ef648d597e5e7afd42636d4cd4608d7c0/Modules/Filters_Layers.bas#L347

A good default percentage is 0.05; higher values will produce a stronger correction.  The function should be simple to reproduce elsewhere (but please mind [the project license](https://github.com/tannerhelland/PhotoDemon/blob/master/LICENSE.md)).

Hope that helps!
